### DISCUSSION TITLE
Space Teams Crashing (Unreal Engine Error)

### DISCUSSION THREAD

**Discussion by Pranayomg** [SimDynamX/SpaceTeamsPRO#41] [2024 Lunar SAR Competition Q&A] [answered]
Hello,
I am having trouble running Space Teams as it crashes when I try to open it.
I have attached an image of the error message (from Unreal Engine):
![error](https://github.com/user-attachments/assets/1f00d3d4-4c68-4ecb-a24d-71831a5dc9b6)

Any help or advice would be appreciated!

**Comment by connorjak** [comment]
I am sorry you are encountering this error. I haven't seen this particular error before; please help us troubleshoot by responding with these pieces of info:
- Full system specifications (use Speccy or a similar tool to create a text file of your system specifications, and drag-and-drop that file into your response)
- Use Task Manager to monitor the performance of your computer while opening Space Teams, and let us know if anything is getting close to 100% such as Memory usage or GPU Memory usage

Some possible resolution steps:
- Ensure your computer meets minimum specifications (System Requirements on the wiki)
- Update your graphics drivers
- Ensure Space Teams is attempting to use your Dedicated GPU when opening (Windows option)

**Reply by connorjak** [reply]
And a PSA for everyone: we do not receive anything when you press submit from the Crash Reporter. This tool is set up by Unreal Engine to send to Epic Games, and it is complicated to change this. Please instead use the Q&A discussion category or git issues to report crashes.

**Reply by Pranayomg** [reply]
> I am sorry you are encountering this error. I haven't seen this particular error before; please help us troubleshoot by responding with these pieces of info:
> 
> * Full system specifications (use Speccy or a similar tool to create a text file of your system specifications, and drag-and-drop that file into your response)
> * Use Task Manager to monitor the performance of your computer while opening Space Teams, and let us know if anything is getting close to 100% such as Memory usage or GPU Memory usage
> 
> Some possible resolution steps:
> 
> * Ensure your computer meets minimum specifications (System Requirements on the wiki)
> * Update your graphics drivers
> * Ensure Space Teams is attempting to use your Dedicated GPU when opening (Windows option)

Hello Connor. Thank you for your response.
I have attached the PC specifications as requested and the GPU/memory usage are not spiking to 100% as I check the Task Manager.
[pc specs.txt](https://github.com/user-attachments/files/17535629/pc.specs.txt)

I have also ensured that the SpaceCRAFT application (SpaceCRAFT.exe) is utilizing the dedicated GPU in the Graphics Settings.

**Reply by connorjak** [reply]
```
NVIDIA GeForce RTX 4060 Laptop GPU
			Manufacturer	NVIDIA
			Model	GeForce RTX 4060 Laptop GPU
			Device ID	10DE-28A0
			Revision	A2
			Subvendor	HP (103C)
			Current Performance Level	Level 0
			Current GPU Clock	2250 MHz
			Current Memory Clock	8000 MHz
			Current Shader Clock	8000 MHz
			Voltage	0.935 V
			Technology	5 nm
			Bus Interface	PCI Express x8
			Temperature	42 °C
			Driver version	32.0.15.5594
			BIOS Version	95.07.28.00.59
			Physical Memory	4091 MB
			Virtual Memory	4092 MB
```

Something appears to be going on here with the RTX 4060 Mobile in your laptop; it appears to Speccy that it has 4GB of VRAM, where typical RTX 4060 Mobiles have 8GB. Could you reply with a screenshot of your `Task Manager`->`Performance`->`GPU (NVidia ...)` page?

**Reply by Pranayomg** [reply]
> ```
> NVIDIA GeForce RTX 4060 Laptop GPU
> 			Manufacturer	NVIDIA
> 			Model	GeForce RTX 4060 Laptop GPU
> 			Device ID	10DE-28A0
> 			Revision	A2
> 			Subvendor	HP (103C)
> 			Current Performance Level	Level 0
> 			Current GPU Clock	2250 MHz
> 			Current Memory Clock	8000 MHz
> 			Current Shader Clock	8000 MHz
> 			Voltage	0.935 V
> 			Technology	5 nm
> 			Bus Interface	PCI Express x8
> 			Temperature	42 °C
> 			Driver version	32.0.15.5594
> 			BIOS Version	95.07.28.00.59
> 			Physical Memory	4091 MB
> 			Virtual Memory	4092 MB
> ```
> 
> Something appears to be going on here with the RTX 4060 Mobile in your laptop; it appears to Speccy that it has 4GB of VRAM, where typical RTX 4060 Mobiles have 8GB. Could you reply with a screenshot of your `Task Manager`->`Performance`->`GPU (NVidia ...)` page?

Here is the image of the GPU in Task Manager:
![image](https://github.com/user-attachments/assets/ca2d5849-8c93-4541-a40d-e869a2e29522)

**Reply by connorjak** [reply]
Oh apparently this is a known issue with Speccy; I don't doubt that your GPU VRAM is 8GB. https://www.reddit.com/r/buildapc/comments/uu7kgy/speccy_only_showing_half_of_the_memory_the_gpu_is/

I'm now referencing this page for possible troubleshooting: https://dev.epicgames.com/documentation/en-us/unreal-engine/how-to-fix-a-gpu-driver-crash-when-using-unreal-engine?application_version=5.3

There also may be ways to manually edit config files or specify command line args to reduce render quality settings.

**Reply by connorjak** [reply]
Here is my opinion of troubleshooting steps to take from here:
1. Please send a zipped copy of your `AppData/Local/SpaceCRAFT/Saved` folder to support@simdynamx.com (as this folder has a possibility of containing data that is personal information). This will allow me to see all the crash files.
2. Update graphics drivers if they are more than ~3 months out of date.
3. Attempt the "How to Resolve TDR Events" section on this page: https://dev.epicgames.com/documentation/en-us/unreal-engine/how-to-fix-a-gpu-driver-crash-when-using-unreal-engine?application_version=5.3 and try launching Space Teams again.
4. Run SpaceCRAFT.exe once with the `-gpucrashdebugging` command line arg then once with `-d3ddebug`, and send your `Saved` folder contents again..

**Comment by connorjak** [comment]
Re-posting here as the reply chain is now long:

Here is my opinion of troubleshooting steps to take from here:
1. Please send a zipped copy of your `AppData/Local/SpaceCRAFT/Saved` folder to support@simdynamx.com (as this folder has a possibility of containing data that is personal information). This will allow me to see all the crash files.
2. Update graphics drivers if they are more than ~3 months out of date.
3. Attempt the "How to Resolve TDR Events" section on this page: https://dev.epicgames.com/documentation/en-us/unreal-engine/how-to-fix-a-gpu-driver-crash-when-using-unreal-engine?application_version=5.3 and try launching Space Teams again.
4. Run SpaceCRAFT.exe once with the `-gpucrashdebugging` command line arg then once with `-d3ddebug`, and send your `Saved` folder contents again.

**Reply by Pranayomg** [reply, marked as answer]
> Re-posting here as the reply chain is now long:
> 
> Here is my opinion of troubleshooting steps to take from here:
> 
> 1. Please send a zipped copy of your `AppData/Local/SpaceCRAFT/Saved` folder to [support@simdynamx.com](mailto:support@simdynamx.com) (as this folder has a possibility of containing data that is personal information). This will allow me to see all the crash files.
> 2. Update graphics drivers if they are more than ~3 months out of date.
> 3. Attempt the "How to Resolve TDR Events" section on this page: https://dev.epicgames.com/documentation/en-us/unreal-engine/how-to-fix-a-gpu-driver-crash-when-using-unreal-engine?application_version=5.3 and try launching Space Teams again.
> 4. Run SpaceCRAFT.exe once with the `-gpucrashdebugging` command line arg then once with `-d3ddebug`, and send your `Saved` folder contents again.

Thank you for the suggestions Connor.
I actually found the issue, it's the NVIDIA driver. All I had to do was disable it in Device Manager under Display Adapters.
The program runs fine now.
